compiler/optimizations/localizeGlobals.cpp

```diff
@@ -69,6 +69,14 @@ void localizeGlobals() {
             fn->insertAtHead(new CallExpr(PRIM_MOVE, local_global, var));
             fn->insertAtHead(new DefExpr(local_global));
 
+            // Copy string immediates to localized strings so that
+            // we can show the string value in comments next to uses.
+            if (VarSymbol* localVarSym = toVarSymbol(var))
+              if (Immediate* immediate = localVarSym->immediate)
+                if (immediate->const_kind == CONST_KIND_STRING)
+                  local_global->immediate =
+                    new Immediate(immediate->v_string, immediate->string_kind);
```

Comment:
I think 'Immediate' objects can be shared. Do we ever deallocate or modify them? I.e. how about

local_global->immediate = localVarSym->immediate

?